dist/tools/travis-scripts/build_and_test.sh

```diff
@@ -9,39 +9,75 @@
 
 set -e
 
+set_result() {
+    NEW_RESULT=$1
+    LAST_RESULT=$2
+
+    if [ $LAST_RESULT -ne 0 ] && [ $NEW_RESULT -eq 0 ]
+    then
+        NEW_RESULT=$LAST_RESULT
+    fi
+
+    echo $NEW_RESULT
+}
+
 if [[ $BUILDTEST_MCU_GROUP ]]
 then
-	if [ "$BUILDTEST_MCU_GROUP" == "static-tests" ]
-	then
-		make -s -C ./examples/default info-concurrency
-		git rebase riot/master || git rebase --abort
-
-		./dist/tools/whitespacecheck/check.sh master || exit
-
-		./dist/tools/licenses/check.sh master --diff-filter=MR --error-exitcode=0 || exit
-		./dist/tools/licenses/check.sh master --diff-filter=AC || exit
-
-		./dist/tools/doccheck/check.sh master || exit
-
-		./dist/tools/externc/check.sh master || exit
-
-		# TODO:
-		#   Remove all but `master` parameters to cppcheck (and remove second
-		#   invocation) once all warnings of cppcheck have been taken care of
-		#   in master.
-		./dist/tools/cppcheck/check.sh master --diff-filter=MR --error-exitcode=0 || exit
-		./dist/tools/cppcheck/check.sh master --diff-filter=AC || exit
-		./dist/tools/pr_check/pr_check.sh riot/master
-		exit 0
-	fi
-	if [ "$BUILDTEST_MCU_GROUP" == "x86" ]
-	then
-
-		make -C ./tests/unittests all test BOARD=native || exit
-		# TODO:
-		#   Reenable once https://github.com/RIOT-OS/RIOT/issues/2300 is
-		#   resolved:
-		#   - make -C ./tests/unittests all test BOARD=qemu-i386 || exit
-	fi
-	./dist/tools/compile_test/compile_test.py
+    if [ "$BUILDTEST_MCU_GROUP" == "static-tests" ]
+    then
+        RESULT=0
+
+        make -s -C ./examples/default info-concurrency
+        git rebase riot/master || git rebase --abort
+
+        RESULT=$(set_result $? $RESULT)
```

Comment:
I would really like it if the RESULT setting would go next to the line that yields the result (i.e. no blank line in between).